drivers/flash/flash_nxp_s32_qspi.c

```diff
@@ -0,0 +1,274 @@
+/*
+ * Copyright 2023-2024 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/flash.h>
+#include <zephyr/logging/log.h>
+
+#include <Qspi_Ip.h>
+
+#include "flash_nxp_s32_qspi.h"
+
+LOG_MODULE_REGISTER(flash_nxp_s32_qspi, CONFIG_FLASH_LOG_LEVEL);
+
+static ALWAYS_INLINE bool area_is_subregion(const struct device *dev, off_t offset, size_t size)
+{
+	Qspi_Ip_MemoryConfigType *memory_cfg = get_memory_config(dev);
+
+	return ((offset >= 0) && (offset < memory_cfg->memSize) &&
+		((memory_cfg->memSize - offset) >= size));
+}
+
+uint8_t nxp_s32_qspi_register_device(void)
+{
+	static uint8_t instance_cnt;
+
+	return instance_cnt++;
+}
+
+/* Must be called with lock */
+int nxp_s32_qspi_wait_until_ready(const struct device *dev)
+{
+	struct nxp_s32_qspi_data *data = dev->data;
+	Qspi_Ip_StatusType status;
+	uint32_t timeout = 0xFFFFFF;
+	int ret = 0;
+
+	do {
+		status = Qspi_Ip_GetMemoryStatus(data->instance);
+		timeout--;
+	} while ((status == STATUS_QSPI_IP_BUSY) && (timeout > 0));
```

Comment:
You really want to busy loop here? Is there any idling in Qspi_Ip_GetMemoryStatus ?

Reply:
there is not waiting timeout in `Qspi_Ip_GetMemoryStatus ` so it is needed.